src/reynir_correct/pattern.py

```diff
@@ -1204,9 +1222,16 @@ def mood_sub_temp(self, match: SimpleTree) -> None:
         so = vp.first_match("so")        
         if so is None: return
         start, end = so.span
-        variants = [f for f in so.all_variants if f != "fh"]
-        variants.append("vh")
+        # Check if so is in a different subclause
+        if "þt" in so.all_variants:
+            return
+        if self.wrong_subtree(match, so):
+            return
+        variants = [f for f in so.all_variants if f != "vh"]
```

Comment:
This could probably be simplified to `variants = list(so.all_variants - {"vh"} | {"fh"})`

Reply:
all_variants returns a list, should I change that to set/frozenset? I only see references in GreynirCorrect and a couple in GreynirPackage so it shouldn't be to difficult/dangerous.

Reply:
This is probably fine.